xcdat/regridder/regrid2.py

```diff
@@ -309,35 +220,35 @@ def _map_latitude(src: xr.DataArray, dst: xr.DataArray) -> Tuple[List, List]:
     src_south, src_north = _extract_bounds(src)
     dst_south, dst_north = _extract_bounds(dst)
 
-    mapping = []
-    weights = []
-
-    for i in range(dst.shape[0]):
-        contrib = np.where(
-            np.logical_and(src_south < dst_north[i], src_north > dst_south[i])
-        )[0]
+    dst_length = dst_south.shape[0]
 
-        mapping.append(contrib)
+    mapping = [
+        np.where(np.logical_and(src_south < dst_north[x], src_north > dst_south[x]))[0]
+        for x in range(dst_length)
+    ]
 
-        north_bounds = np.minimum(dst_north[i], src_north[contrib])
-        south_bounds = np.maximum(dst_south[i], src_south[contrib])
+    bounds = [
+        (np.minimum(dst_north[x], src_north[y]), np.maximum(dst_south[x], src_south[y]))
+        for x, y in enumerate(mapping)
+    ]
 
-        weight = np.sin(np.deg2rad(north_bounds)) - np.sin(np.deg2rad(south_bounds))
-
-        weights.append(weight.values.reshape(contrib.shape[0], 1))
+    weights = [
+        (np.sin(np.deg2rad(x)) - np.sin(np.deg2rad(y))).reshape((-1, 1))
+        for x, y in bounds
+    ]
```

Comment:
Can we add comments about the logic just in case for future developer reference? It'll take me time to analyze and understand the code without it.

Reply:
Fixed in https://github.com/xCDAT/xcdat/pull/533/commits/96eaae21ebdf20fe51831bca24a8a892a6ed6cf0

Reply:
Resolved with https://github.com/xCDAT/xcdat/pull/533/commits/7695e2b89f16a430045c7e04f12c89fd1edbcdc6